### PR TITLE
[BUGFIX] Prevent overwriting source files

### PR DESCRIPTION
**Pull request by hhoechtl**
After switching from FileSystemSymlinkTarget to FileSystemTarget
the symlinks are still present in the targetPath and therefore Flow
tries to copy the stream content in the symlink (original file, same
stream) which results in a 0 byte file.
We handle this issue by removing the symlinks if present.

Fixes: FLOW-384

### CODE REVIEW

**Comment by aertmann** [comment]
needs to be rebased

**Comment by hhoechtl** [comment]
Done

**Comment by kitsunet** [comment]
lol, we are to slow... Can we agree on the usefulness of this and then rebase and merge?
I totally see the problem and think we need it, even though I don't like that there is so much new code to fix it. I also don't see another way. So for now :+1:

File: TYPO3.Flow/Classes/TYPO3/Flow/Resource/Target/FileSystemTarget.php [hunk 1]
```diff
@@ -151,6 +151,25 @@ public function getName()
     }
 
     /**
+     * Checks if the PackageStorage has been previously initialized with symlinks
+     * and clears them. Otherwise the original sources would be overwritten.
+     *
+     * @param \TYPO3\Flow\Resource\Storage\StorageInterface $storage
```
Comment (aertmann) [inline TYPO3.Flow/Classes/TYPO3/Flow/Resource/Target/FileSystemTarget.php]: @return void

**Comment by albe** [comment]
:+1: but what about the other way round? Switching from FileSystemTarget to FileSystemSymlinkTarget? Would that just work?

**Comment by hhoechtl** [comment]
Sorry for the long delay. Yes, the switch from FileSystemTarget to FileSystemSymlinkTarget worked as far as I can remember.

**Comment by hhoechtl** [comment]
Reopened pull request https://github.com/neos/flow-development-collection/pull/196

**Pull request closed without merging**